File: Modules/SEDO/EMS/SIE.Web.EMS/Tpms/Commands/LookImageCommand.cs
using SIE.Web.Command;

namespace SIE.Web.EMS.Tpms.Commands
{
    /// <summary>
    /// 查看图片命令
    /// </summary>
    [JsCommand("SIE.Web.EMS.Tpms.Commands.LookImageCommand")]
    public class LookImageCommand : ViewCommand<ViewArgs>
    {
        /// <summary>
        /// 查看图片
        /// </summary>
        /// <param name="args">args</param>
        /// <param name="scope">scope</param>
        /// <returns>结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            return true;
        }
    }
}

File: Modules/SEDO/EMS/SIE.Web.EMS/Tpms/Commands/LookImageCommand.js
SIE.defineCommand('SIE.Web.EMS.Tpms.Commands.LookImageCommand', {
    extend: 'SIE.cmd.Edit',
    meta: { text: "查看图片", group: "ImageEdit" },
    canExecute: function (view) {
        var curr = view.getCurrent();
        if (curr && curr.getPhoto().length > 0) {
            return true;
        }
        return false;
    },
    execute: function (view, source) {
        var me = this;
        var indata = {};
        var Insp = view.getSelection()[0].getData();
        indata.Data = Ext.encode({ Insp: Insp });
        if (Insp.Photo != undefined || Insp.Photo != null) {
            var win = SIE.Window.show({
                title: '查看图片'.t(),
                width: '50%',
                height: '50%',
                items: [
                Ext.create('Ext.Img', {
                    height: '100%',
                    width: '100%',
                    src: Insp.Photo
                })
                ],
                buttons: []
            });
        }
    }
});